File: src/BaseHmiTypes/Screens/Base/HmiVariableConfirmationSettings.cs
namespace BaseHmiTypes.Screens.Base;

using BaseHmiTypes.Screens.Widgets;

public sealed class HmiVariableConfirmationSettings
{
    public bool? Enabled { get; set; }

    public string? Selector { get; set; }

    public double? ConfirmationValue { get; set; }

    public double? FirstSignatureValue { get; set; }

    public double? SecondSignatureValue { get; set; }

    public HmiConfirmationDialogSettings? ConfirmationDialog { get; set; }

    public HmiElectronicSignatureSettings? FirstSignature { get; set; }

    public HmiElectronicSignatureSettings? SecondSignature { get; set; }
}

File: typescript/base-hmi-types/src/screens/base/HmiVariableConfirmationSettings.ts
import { HmiConfirmationDialogSettings } from "../widgets/HmiConfirmationDialogSettings.js";
import { HmiElectronicSignatureSettings } from "./HmiElectronicSignatureSettings.js";

export class HmiVariableConfirmationSettings {
  enabled?: boolean;
  selector?: string;
  confirmationValue?: number;
  firstSignatureValue?: number;
  secondSignatureValue?: number;
  confirmationDialog?: HmiConfirmationDialogSettings;
  firstSignature?: HmiElectronicSignatureSettings;
  secondSignature?: HmiElectronicSignatureSettings;
}
